networkx/utils/backends.py

```diff
@@ -541,8 +558,31 @@ def __call__(self, /, *args, backend=None, **kwargs):
             )
 
         # Only networkx graphs; try to convert and run with a backend with automatic
-        # conversion, but don't do this by default for graph generators or loaders.
-        if self.graphs:
+        # conversion, but don't do this by default for graph generators or loaders,
+        # or if the functions mutates an input graph.
+        if (
+            self.graphs
+            and (
+                not self.mutates_input
+                or isinstance(self.mutates_input, dict)
+                # If `mutates_input` begins with "not ", then assume the argument is boolean,
+                # otherwise treat it as a node or edge attribute if it's not None.
+                and any(
+                    not (
+                        args[arg_pos]
+                        if len(args) > arg_pos
+                        else kwargs.get(arg_name[4:], True)
```

Comment:
Note that this implicitly assumes e.g. `copy=True` is the default everywhere (which is true today).